apps/bluetooth/js/modules/pair_manager.js

```diff
@@ -38,21 +62,176 @@ define(function(require) {
           this.showPendingPairing(event.settingValue);
       }.bind(this));
 
-      // Observe Bluetooth ondisabled event from hardware side.
+      // Observe Bluetooth state from hardware side.
       // Then, close pairing dialog immediately.
-      navigator.mozBluetooth.ondisabled = this.onBluetoothDisabled.bind(this);
+      BtContext.observe('state', (state) => {
+        if (state !== 'enabled') {
+          this.onBluetoothDisabled();
+        }
+      });
 
       // Observe Bluetooth enabled state for close pairing dialog immediately.
-      navigator.mozSettings.addObserver('bluetooth.enabled',
-        function gotBluetoothEnabledChanged(event) {
-          if (!event.settingValue) {
-            this.onBluetoothDisabled();
-          }
-      }.bind(this));
+      navigator.mozSettings.addObserver('bluetooth.enabled', (event) => {
```

Comment:
This is not required as we already observe the bluetooth state change using `BtContext`.